File: 0680-valid-palindrome-ii/0680-valid-palindrome-ii.cpp
class Solution {
public:
    bool check(int i,int j,string &s){
        while(i<j){
            if(s[i]!=s[j]) return false;
            i++;
            j--;
        }
        return true;
    }
    bool validPalindrome(string s) {
        int i=0;
        int j=s.size()-1;

        while(i<j){
            if(s[i]!=s[j]){
                return check(i,j-1,s) || check(i+1,j,s);
            }
            i++;
            j--;
        }

        return true;
    }
};

File: 0680-valid-palindrome-ii/0680-valid-palindrome-ii.java
class Solution { 
    public boolean solve(String s,int i,int j){
        while(i<=j){
            if(s.charAt(i)!=s.charAt(j)) return false;
            i++;
            j--;
        }
        return true;
    }

    public boolean validPalindrome(String s) {
        int i=0;
        int j=s.length()-1;

        while(i<=j){
            if(s.charAt(i)!=s.charAt(j)){
                return solve(s,i+1,j) || solve(s,i,j-1);
            }
            i++;
            j--;    
        }

        return true;
    }
}